### DISCUSSION TITLE
window only display if initial children is not empty

### DISCUSSION THREAD

**Discussion by deng232** [Aylur/ags#431] [Q&A] [answered]
I'm try to write my own notification window and I found that the window only display when the initial Box children is not empty. Looking at the  inspector, the children Box have created successfully whenever a notification was being sent, it just nothing displayed nothing whenever the Box children was initially empty.
```

const test = Widget.Label("test")


const Nw = Widget.Window({
    name:"notify-win",
    child: Widget.Box({
        vertical: true,
        children: [test] as Gtk.Widget[] // this will work fine, but if left it empty [], nothing will being display when notification sent
    })
})


notifications.connect("notified", (n, id) => {
    const note = n.getNotification(id)
    if (note === undefined) {
        throw Error("non-exists notification")
    }
    print(`called ${note.summary}`)
    Nw.child.children = [nBox(note), ...Nw.child.children] // nBox return a EventBox 
})

export default Nw
```

**Comment by kotontrion** [comment, marked as answer]
this is a common issue with gtk, as listed [here in wiki](https://aylur.github.io/ags-docs/config/common-issues/#window-doesnt-show-up)

The window won't show up when it has initially no size, so when you initialize it with no children you have to force it to have a size, which can be achieved using css.

```js
const Nw = Widget.Window({
    name:"notify-win",
    child: Widget.Box({
        vertical: true,
        css: "padding: 1px;", //window has now a minimum size of 2x2 px and therefore show up.
        children: []
    })
})
```
